src/core/net/udp6.cpp

```diff
@@ -74,7 +74,12 @@ bool Udp::SocketHandle::Matches(const MessageInfo &aMessageInfo) const
 Udp::Socket::Socket(Instance &aInstance)
     : InstanceLocator(aInstance)
 {
-    mHandle = nullptr;
+    memset(&mSockName, 0, sizeof(mSockName));
+    memset(&mPeerName, 0, sizeof(mPeerName));
+    mHandler = nullptr;
+    mContext = nullptr;
+    mHandle  = nullptr;
+    mNext    = nullptr;
```

Comment:
```suggestion
    Clear();
```
Let `Socket` inherit from `Clearable` ?

Reply:
I don't think we can have `Socket` be `Clearable` sicne it is `InstanceLocator`.
`Clear()` would `memset` all the fields to zero which we should not in this case (should not write over the instance ref/pointer).

Reply:
Another approach here: 
- We can have `SocketHandle` be a `Clearable` (handle just contains the fields and mirrors the public OT `otUdpSocket` struct). 
- Then `Socket` would inherit the `Clear()` method from `SocketHandle` and can use it in its own constructor (which won't touch any fields potentially added by `InstanceLocator`).

Reply:
Done, thanks.